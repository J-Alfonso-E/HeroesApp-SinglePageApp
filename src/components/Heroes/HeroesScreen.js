import { Navigate, useParams } from "react-router-dom"
import { getHeroById } from "../../selectors/getheroById";



export const HeroesScreen = () => {

    const {HeroId} = useParams(); //Leer Argumento que viene en el enlace del DashboardRoutes.js

    
    const hero = getHeroById(HeroId);
    if(!hero) return <Navigate to="/" />

    return (
        <div>
            <h1>HeroesScreen</h1>

            <p>{hero.superhero}</p>
        </div>
    )
}