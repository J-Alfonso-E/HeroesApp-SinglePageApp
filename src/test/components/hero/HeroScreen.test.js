import { mount } from 'enzyme';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { HeroesScreen } from '../../../components/Heroes/HeroesScreen';


const mockNavigate = jest.fn(); //Funcion de prueba de useNavigate, forzosamente se requiere que comience con "mock"

jest.mock('react-router-dom', () => ({ //Simulacion de la libreria de de react-router-dom
    ...jest.requireActual('react-router-dom'), //Importa la libreria de react sin cambios
    useNavigate: () => mockNavigate //Despues de que se importo se sobreescribe el hook de useNavigate
}));


describe('Pruebas en heroScreen', () => {

    test("No debe de mostrar el HeroScreen si no hay un heroe en la URL", () => {
        const wrapper = mount(
        <MemoryRouter initialEntries={['/hero']} >
        <Routes>
            <Route path="/hero" element={<HeroesScreen />} />
            <Route path="/" element={<h1>Sin Contenido</h1>} />
        </Routes>
    </MemoryRouter>
    );
        

        expect(wrapper.find('h1').text().trim()).toBe('Sin Contenido');
    });

    test("Debe de mostrar un heroe", () => {
        const wrapper = mount(
        <MemoryRouter initialEntries={['/hero/marvel-captain']} >
        <Routes>
            <Route path="/hero/:HeroId" element={<HeroesScreen />} />
            <Route path="/" element={<h1>Sin Contenido</h1>} />
        </Routes>
    </MemoryRouter>
    );
        
            expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.row').exists()).toBe(true);
    });

    test("Verificando que regrese a la pantalla anterior", () => {
        const wrapper = mount(
        <MemoryRouter initialEntries={['/hero/marvel-captain']} >
        <Routes>
            <Route path="/hero/:HeroId" element={<HeroesScreen />} />
            
        </Routes>
    </MemoryRouter>
    );
        
        wrapper.find('button').simulate('click');
        expect(mockNavigate).toHaveBeenCalledWith(-1);
    });


    test("No debe de mostrar el HeroScreen si la ruta esta mal", () => {
        const wrapper = mount(
        <MemoryRouter initialEntries={['/hero/dfcasca']} >
        <Routes>
            <Route path="/hero/:HeroId" element={<HeroesScreen />} />
            <Route path="/" element={<h1>Sin Contenido</h1>} />
        </Routes>
    </MemoryRouter>
    );
        

        expect(wrapper.find('h1').text().trim()).toBe('Sin Contenido');
    });
});