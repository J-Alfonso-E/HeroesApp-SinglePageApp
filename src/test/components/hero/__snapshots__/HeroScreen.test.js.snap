// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en heroScreen Debe de mostrar un heroe 1`] = `
<div>
  <div
    className="row mt-5"
  >
    <div
      className="col-4 animate__animated animate__bounceInLeft"
    >
      <img
        alt="Captain America"
        className="img-thumbnail"
        src="/assets/marvel-captain.jpg"
      />
    </div>
    <div
      className="col-8"
    >
      <h3>
        Captain America
      </h3>
      <ul
        className="list-group animate__animated animate__bounceInRight"
      >
        <li
          className="list-group-item"
        >
           
          <b>
            Alter Ego:
          </b>
           
          Steve Rogers
        </li>
        <li
          className="list-group-item"
        >
           
          <b>
            Publisher:
          </b>
           
          Marvel Comics
        </li>
        <li
          className="list-group-item"
        >
           
          <b>
            First Appearance:
          </b>
           
          Captain America Comics #1
        </li>
      </ul>
      <button
        className="btn btn-outline-info"
        onClick={[Function]}
      >
        Regresar
      </button>
    </div>
  </div>
</div>
`;
